functions.py

```diff
@@ -0,0 +1,44 @@
+import os
+import json
+import requests
+import urllib.request
+from datetime import datetime
+
+from typing import Dict
+
+
+def getFilenames(path: str) -> list:
+    "Get names of all files in the given directory"
+    for (_, _, filenames) in os.walk(path):
+        if filenames:
+            return filenames
+        return []
+    raise ValueError("Incorrect path given")
+
+def saveImage(path: str, filename: str, url: str) -> None:
+    """Save image from URL to /path/filename.ext
+    
+    File extension is taken from URL
+    """
+    fileExtension = url.split('.')[-1]
+    image = requests.get(url).content
+    with open(f'{path}{filename}.{fileExtension}', 'wb') as newImage:
```

Comment:
the safer way to do it is [os.path.join](https://docs.python.org/3/library/os.path.html#os.path.join) - this works on all operating systems and handles the `/` or `\` for you :)